openslides_backend/action/action.py

```diff
@@ -134,29 +136,94 @@ def perform(
         final_write_request = self.process_write_requests()
         return (final_write_request, results)
 
-    def check_permissions(self, payload: ActionData) -> None:
+    def check_permissions(self, instance: Dict[str, Any]) -> None:
         """
-        Checks permission by requesting permission service.
+        Checks permission by requesting permission service or using internal check.
         """
-        if not self.permission.is_allowed(self.name, self.user_id, list(payload)):
-            raise PermissionDenied(
-                f"You are not allowed to perform action {self.name}."
+        # switch between internal and external permission service
+        if self.permission:
+            meeting_id = self.get_meeting_id(instance)
+            if self.has_perm(self.permission, meeting_id):
+                return
+        else:
+            if self.permission_service.is_allowed(self.name, self.user_id, [instance]):
+                return
+
+        raise PermissionDenied(f"You are not allowed to perform action {self.name}.")
+
+    def get_meeting_id(self, instance: Dict[str, Any]) -> int:
+        """
+        To be overridden in subclasses. Must return the meeting_id that belongs to this instance.
+        """
+        raise NotImplementedError()
+
+    def has_perm(self, permission: str, meeting_id: int) -> bool:
```

Comment:
I think the logic is right. IMO you should ass a new auth module (or permission module) with these kinds of functions. One additional thought. You can include `enable_anonymous` and `default_group_id` into the first call to the datastore.

Reply:
done